src/context/Authentication/AuthState.js

```diff
@@ -56,14 +71,28 @@ const AuthState = ({ children }) => {
       if (res.status === 200) {
         console.log(res.data);
         dispatch({
-          type: 'SET_USER',
+          type: "SET_USER",
           payload: res.data,
         });
       }
     });
   };
 
-  const setLoading = () => dispatch({ type: 'SET_LOADING' });
+  const logOut = () => {
+    AsyncStorage.removeItem("token");
+    dispatch({
+      type: "LOGOUT",
+    });
+  };
+
+  const toggleTheme = async (isDark) => {
+    const theme = isDark ? darkTheme : lightTheme;
+    dispatch({
+      type: "SET_THEME",
+      payload: theme,
+    });
+    AsyncStorage.setItem("themeMode", theme.themeMode);
+  };
```

Comment:
What do you think about creating a separate context (Theme/themeContext.js ThemeState.js themeReducer.js ) for the theme?